Makefile

```diff
@@ -1,22 +1,40 @@
 
 NAME = floop
 FILES = cmd/main.go cmd/cli.go
+BRANCH = $(shell git rev-parse --abbrev-ref HEAD)
+COMMIT = $(shell git rev-parse --short HEAD)
+VERSION = $(shell git describe | sed -e "s/^v//")
+BUILDTIME = $(shell date +%Y-%m-%dT%T%z)
+BUILD_CMD = CGO_ENABLED=0 go build -a -tags netgo -installsuffix netgo \
+	-ldflags="-X main.branch=${BRANCH} -X main.commit=${COMMIT} -X main.buildtime=${BUILDTIME} -w"
 
-BUILD_CMD = go build
+version:
+	@echo $(VERSION)
 
 clean:
 	go clean -i ./...
-	rm -f $(NAME)
-	rm -rf ./dist
+	rm -rf $(NAME) ./dist coverage.out
 
 deps:
 	go get -d ./...
 
-$(NAME):
-	go build -o $(NAME) $(FILES)
+test:
+	go test -v -coverprofile=coverage.out .
 
-dist:
+${NAME}:
+	$(BUILD_CMD) -o $(NAME) $(FILES)
+
+dist: clean
 	[ -d ./dist ] || mkdir ./dist
-	GOOS=linux $(BUILD_CMD) -o ./dist/$(NAME)-linux $(FILES)
-	GOOS=darwin $(BUILD_CMD) -o ./dist/$(NAME)-darwin $(FILES)
-	GOOS=windows $(BUILD_CMD) -o ./dist/$(NAME)-windows $(FILES)
+
+	$(eval OS := linux) $(eval OUTFILE := ./dist/$(NAME)-$(OS)-$(VERSION))
```

Comment:
Evals are not required for this.  You can simple compile each bin with `GOOS=linux|darwin|windows`